### PR TITLE
Refactor code and add new features

### PR DESCRIPTION
**Pull request by kpqdap**
This pull request includes various code refactorings and the addition of new features. It also updates package dependencies and fixes some bugs.

### CODE REVIEW

File: src/next.config.js [hunk 1]
```diff
@@ -67,13 +64,16 @@ const nextConfig = {
...
     },
   },
+  devIndicators: {
+    buildActivityPosition: "bottom-right",
+  },
   experimental: {
     serverComponentsExternalPackages: ["@azure/storage-blob"],
     serverActions: {
-      allowedOrigins: [localUrl, fullUrl],
+      allowedOrigins: ["*.ai.qld.gov.au", "qggptprodopenai.azurewebsites.net", "qggptdevopenai.azurewebsites.net"],
```
Comment (QDAP-Fred) [inline src/next.config.js:73]: can you add these in `.env` variables instead?

File: src/next.config.js [hunk 2]
```diff
@@ -27,7 +24,7 @@ const securityHeaders = [
   {
     key: "Content-Security-Policy",
     value:
-      "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-ancestors 'self'; img-src 'self'; font-src 'self' data:; connect-src 'self' https://qdap-dev-apim.azure-api.net https://australiaeast-1.in.applicationinsights.azure.com/; media-src 'self'; frame-src 'self'; object-src 'none';",
+      "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-ancestors 'self'; img-src 'self'; font-src 'self' data:; connect-src 'self' https://qdap-dev-apim.azure-api.net https://qdap-prd-apim.developer.azure-api.net *.ai.qld.gov.au https://australiaeast-1.in.applicationinsights.azure.com/ https://australiaeast.livediagnostics.monitor.azure.com/; media-src 'self'; frame-src 'self'; object-src 'none';",
```
Comment (QDAP-Fred) [inline src/next.config.js:27]: can you add these in .env variables instead?

File: src/app/api/chat/route.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 
-const delay = (ms: number | undefined): Promise<void> => new Promise(resolve => setTimeout(resolve, ms))
+const delay = async (ms: number | undefined): Promise<void> => new Promise(resolve => setTimeout(resolve, ms))
 
 const errorMessages: { [key: number]: string } = {
   400: "Oops! Something went wrong with your request.",
   401: "Access denied. Please ensure your credentials are correct.",
   402: "Whoops, looks like you've exceeded your limit! Please try again later.",
   403: "Sorry, we're unable to fulfill your request.",
+  404: "We couldn't find send your message for reply, please try again or report the issue.",
```
Comment (QDAP-Fred) [inline src/app/api/chat/route.ts:10]: Copy to review

**Review by QDAP-Fred — CHANGES_REQUESTED** [review]

File: src/next.config.js (hunk 1 above)
Comment (kpqdap) [reply src/next.config.js:76]: They don't add during build for some reason

**Review by QDAP-Fred — APPROVED** [review]

**Pull request merged**